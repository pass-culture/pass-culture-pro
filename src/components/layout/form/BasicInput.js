import React from 'react'

const BasicInput = props => {
  return <input
    aria-describedby={props['aria-describedby']}
    autoComplete={props.autoComplete}
    checked={props.checked}
    className={`input is-${props.size}`}
    disabled={props.disabled}
    id={props.id}
    name={props.name}
    onChange={props.onChange}
    required={props.required}
    readOnly={props.readOnly}
    type={props.type}
    value={props.value}
  />
}

export default BasicInput
