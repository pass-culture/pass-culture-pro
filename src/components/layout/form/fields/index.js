export * from './DateField'
export * from './HiddenField'
export * from './NumberField'
export * from './TextField'
export * from './TextareaField'
export * from './TimeField'
