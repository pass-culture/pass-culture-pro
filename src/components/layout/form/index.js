export * from './FieldError'
export * from './TextField'
