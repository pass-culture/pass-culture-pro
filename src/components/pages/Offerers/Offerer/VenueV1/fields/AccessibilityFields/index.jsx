export { default, autoFillNoDisabilityCompliantDecorator } from './AccesibilityFields'
