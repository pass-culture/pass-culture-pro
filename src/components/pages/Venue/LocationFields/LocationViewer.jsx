import classnames from 'classnames'
import { Icon as LeafletIcon } from 'leaflet'
import debounce from 'lodash.debounce'
import PropTypes from 'prop-types'
import React, { Component, Fragment } from 'react'
import Autocomplete from 'react-autocomplete'
import { Map, Marker, TileLayer } from 'react-leaflet'

import getSuggestionsFromAddressAndMaxSuggestions from './utils/getSuggestionsFromAddressAndMaxSuggestions'
import getSuggestionsFromLatitudeAndLongitude from './utils/getSuggestionsFromLatitudeAndLongitude'
import sanitizeCoordinates from './utils/sanitizeCoordinates'
import { FRANCE_POSITION } from './utils/positions'
import { ROOT_PATH } from 'utils/config'

const markerIcon = new LeafletIcon({
  iconUrl: `${ROOT_PATH}/icons/ico-geoloc-solid2.svg`,
  iconRetinaUrl: `${ROOT_PATH}/icons/ico-geoloc-solid2.svg`,
  iconSize: [21, 30],
  iconAnchor: [10, 30],
  popupAnchor: null,
})

class LocationViewer extends Component {
  constructor(props) {
    super(props)
    this.state = {
      draggable: true,
      inputValue: '',
      isLoading: false,
      hasSelectedSuggestion: false,
      marker: null,
      position: props.initialPosition,
      suggestions: [],
    }
    this.refmarker = React.createRef()
    this.onDebouncedFetchSuggestions = debounce(
      this.fetchSuggestions,
      props.debounceTimeout
    )
  }

  static getDerivedStateFromProps = (newProps, currentState) => {
    const latitude = sanitizeCoordinates(newProps.latitude)
    const longitude = sanitizeCoordinates(newProps.longitude)

    const hasCoordinates = latitude && longitude

    const nextPosition = {
      latitude: latitude || newProps.defaultInitialPosition.latitude,
      longitude: longitude || newProps.defaultInitialPosition.longitude,
      zoom: hasCoordinates
        ? newProps.zoom
        : newProps.defaultInitialPosition.zoom,
    }
    const isInputValueEmpty = newProps.value === ''
    const nextInputValue = isInputValueEmpty
      ? ''
      : newProps.value || currentState.inputValue
    const nextStateWithPositionAndInputValue = {
      inputValue: nextInputValue,
      position: nextPosition,
    }

    const nextStateWithSuggestionsAndMarker = hasCoordinates
      ? {
          suggestions: currentState.hasSelectedSuggestion
            ? []
            : currentState.suggestions,
          marker: {
            latitude,
            longitude,
          },
        }
      : null

    return Object.assign(
      {},
      currentState,
      nextStateWithPositionAndInputValue,
      nextStateWithSuggestionsAndMarker
    )
  }

  toggleDraggable = () => {
    this.setState({ draggable: !this.state.draggable })
  }

  onMarkerDragend = () => {
    const { readOnly, onMarkerDragend } = this.props
    if (readOnly) {
      return
    }
    const {
      lat: latitude,
      lng: longitude,
    } = this.refmarker.current.leafletElement.getLatLng()

    this.setState({
      marker: {
        latitude,
        longitude,
      },
      suggestions: [],
    })

    getSuggestionsFromLatitudeAndLongitude(latitude, longitude).then(result => {
      this.setState({
        isLoading: false,
      })

      if (result.error) {
        return
      }

      const hasOnlyOneSuggestion = result.data && result.data.length === 1
      if (hasOnlyOneSuggestion) {
        const suggestion = result.data[0]
        const { address, city, postalCode } = suggestion

        const location = {
          address,
          city,
          latitude,
          longitude,
          postalCode,
        }

        onMarkerDragend(location)
        return
      }

      const location = {
        address: null,
        city: null,
        latitude,
        longitude,
        postalCode: null,
      }

      onMarkerDragend(location)
    })
  }

  onTextChange = event => {
    const { onTextChange: onTextChangeFromProps } = this.props
    const address = event.target.value
    this.setState({
      hasSelectedSuggestion: false,
      inputValue: address,
    })

    const location = {
      address,
      city: null,
      latitude: null,
      longitude: null,
      postalCode: null,
    }

    onTextChangeFromProps(location)
    this.onDebouncedFetchSuggestions(address)
  }

  onSuggestionSelect = (address, location) => {
    const { onSuggestionSelect: onSuggestionSelectFromProps, zoom } = this.props
    const { latitude, longitude, placeholder } = location
    if (placeholder) return
    this.setState({
      hasSelectedSuggestion: true,
      inputValue: address,
      position: {
        latitude,
        longitude,
        zoom,
      },
      marker: {
        latitude,
        longitude,
      },
    })
    onSuggestionSelectFromProps(location)
  }

  fetchSuggestions = address => {
    const { maxSuggestions, placeholder } = this.props
    this.setState({ isLoading: true })

    // NOTE: CANNOT EXPRESS THIS WITH AWAIT ASYNC
    // BECAUSE this.props cannot be found in that case...
    // weird
    getSuggestionsFromAddressAndMaxSuggestions(address, maxSuggestions).then(
      result => {
        if (result.error) {
          return
        }

        const hasNoData = result.data.length === 0
        if (hasNoData) {
          this.setState({
            isLoading: false,
          })
          return
        }

        const defaultSuggestion = {
          label: placeholder,
          placeholder: true,
          id: 'placeholder',
        }

        const suggestions = result.data.concat(defaultSuggestion)
        this.setState({
          isLoading: false,
          suggestions,
        })
      }
    )
  }

  renderSuggestionsMenu = children => {
    const empty = children.length === 0
    return <div className={classnames('menu', { empty })}>{children}</div>
  }

  renderSuggestion = ({ id, label, placeholder }, highlighted) => (
    <div
      className={classnames({
        item: true,
        highlighted,
        placeholder,
      })}
      key={id}>
      {label}
    </div>
  )

  renderInput() {
    const {
      className,
      id,
      name,
      placeholder,
      readOnly,
      required,
      value,
    } = this.props
    const { inputValue, isLoading, suggestions } = this.state

    if (readOnly) {
      return (
        <input
          className={className}
          name={name}
          readOnly={readOnly}
          value={inputValue}
        />
      )
    }

    return (
      <Fragment>
        <Autocomplete
          autocomplete="street-address"
          getItemValue={value => value.label}
          inputProps={{
            className: className || `input`,
            id,
            name,
            placeholder,
            readOnly,
            required,
          }}
          items={suggestions}
          onChange={this.onTextChange}
          onSelect={this.onSuggestionSelect}
          readOnly={readOnly}
          renderItem={this.renderSuggestion}
          renderMenu={this.renderSuggestionsMenu}
          value={value || inputValue}
          wrapperProps={{ className: 'input-wrapper' }}
        />
        <button
          className={classnames('button is-loading', {
            'is-invisible': !isLoading,
          })}
        />
      </Fragment>
    )
  }

  render() {
    const { readOnly, withMap } = this.props
    const { marker, position } = this.state

    if (!withMap) return this.renderInput()
    const { latitude, longitude, zoom } = position

    return (
      <div className="location-viewer">
        {this.renderInput()}
        <Map center={[latitude, longitude]} zoom={zoom} className="map">
          <TileLayer url="https://cartodb-basemaps-{s}.global.ssl.fastly.net/light_all/{z}/{x}/{y}.png" />
          {marker && (
            <Marker
              draggable={!readOnly}
              onDragend={this.onMarkerDragend}
              position={[marker.latitude, marker.longitude]}
              icon={markerIcon}
              ref={this.refmarker}
              alt={[marker.latitude, marker.longitude].join('-')}
            />
          )}
        </Map>
      </div>
    )
  }
}

LocationViewer.defaultProps = {
  debounceTimeout: 300,
  defaultInitialPosition: FRANCE_POSITION,
  latitude: null,
  longitude: null,
  maxSuggestions: 5,
  name: null,
  onMarkerDragend: () => {},
  onSuggestionSelect: () => {},
  onTextChange: () => {},
  placeholder: "Sélectionnez l'adresse lorsqu'elle est proposée.",
  withMap: false,
  zoom: 15,
}

LocationViewer.propTypes = {
  debounceTimeout: PropTypes.number,
  defaultInitialPosition: PropTypes.shape({
    latitude: PropTypes.number,
    longitude: PropTypes.number,
    zoom: PropTypes.number,
  }),
  latitude: PropTypes.number,
  longitude: PropTypes.number,
  maxSuggestions: PropTypes.number,
  name: PropTypes.string,
  onMarkerDragend: PropTypes.func,
  onSuggestionSelect: PropTypes.func,
  onTextChange: PropTypes.func,
  placeholder: PropTypes.string,
  withMap: PropTypes.bool,
  zoom: PropTypes.number,
}

export default LocationViewer
