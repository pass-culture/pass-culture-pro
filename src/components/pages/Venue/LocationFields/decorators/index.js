export * from './bindGetSuggestionsToLatitude'
export * from './bindGetSuggestionsToLongitude'
