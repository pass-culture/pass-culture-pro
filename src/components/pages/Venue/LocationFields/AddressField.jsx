/* eslint
  react/jsx-one-expression-per-line: 0 */
import classnames from 'classnames'
import React from 'react'
import PropTypes from 'prop-types'
import { Field } from 'react-final-form'
import { composeValidators } from 'react-final-form-utils'

import LocationViewer from './LocationViewer'
import FieldErrors from 'components/layout/form/FieldErrors'
import getRequiredValidate from 'components/layout/form/utils/getRequiredValidate'

const createBatchLocationChanges = (form, isLocationFrozen) => location => {
  form.batch(() => {
    const { address, city, latitude, longitude, postalCode } = location
    form.change('address', address)
    form.change('city', city)
    form.change('isLocationFrozen', isLocationFrozen)
    form.change('latitude', latitude)
    form.change('longitude', longitude)
    form.change('postalCode', postalCode)
  })
}

export const AddressField = ({
  className,
  disabled,
  form,
  format,
  id,
  innerClassName,
  label,
  name,
  parse,
  placeholder,
  readOnly,
  renderInner,
  renderValue,
  required,
  validate,
  ...AddressProps
}) => (
  <Field
    format={format}
    name={name}
    validate={composeValidators(validate, getRequiredValidate(required))}
    render={({ input, meta }) => {
      return (
        <div
          className={classnames('field text-field', className, {
            'is-label-aligned': label,
            'is-read-only': readOnly,
          })}
          id={id}>
          <label
            htmlFor={name}
            className={classnames('field-label', { empty: !label })}>
            {label && (
              <span>
                <span>{label}</span>
                {required && !readOnly && (
                  <span className="field-asterisk">*</span>
                )}
              </span>
            )}
          </label>
          <div className="field-control">
            <div className="field-value flex-columns items-center">
              <div
                className={classnames(
                  'field-inner flex-columns items-center',
                  innerClassName
                )}>
                <LocationViewer
                  {...input}
                  {...AddressProps}
                  className="field-input field-address"
                  disabled={disabled || readOnly}
                  name={name}
                  onMarkerDragend={createBatchLocationChanges(form, false)}
                  onSuggestionSelect={createBatchLocationChanges(form, true)}
                  onTextChange={createBatchLocationChanges(form, false)}
                  placeholder={readOnly ? '' : placeholder}
                  readOnly={readOnly}
                  required={!!required}
                />
              </div>
            </div>
            <FieldErrors meta={meta} />
          </div>
          <div />
        </div>
      )
    }}
  />
)

AddressField.defaultProps = {
  className: '',
  disabled: false,
  format: null,
  id: null,
  innerClassName: null,
  label: '',
  parse: null,
  placeholder: '',
  readOnly: false,
  required: false,
  type: 'text',
  validate: null,
}

AddressField.propTypes = {
  autoComplete: PropTypes.string,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  form: PropTypes.object.isRequired,
  format: PropTypes.func,
  id: PropTypes.string,
  innerClassName: PropTypes.string,
  label: PropTypes.string,
  name: PropTypes.string.isRequired,
  parse: PropTypes.func,
  placeholder: PropTypes.string,
  readOnly: PropTypes.bool,
  renderInner: PropTypes.func,
  renderValue: PropTypes.func,
  required: PropTypes.oneOfType([PropTypes.bool, PropTypes.func]),
  validate: PropTypes.func,
}

export default AddressField
