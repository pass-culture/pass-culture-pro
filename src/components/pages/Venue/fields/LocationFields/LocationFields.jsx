import PropTypes from 'prop-types'
import React from 'react'

import AddressField from './AddressField'
import {
  HiddenField,
  NumberField,
  TextField,
} from 'components/layout/form/fields'

const LocationFields = ({
  fieldReadOnlyBecauseFrozenFormSiret,
  form,
  formLatitude,
  formLongitude,
  formIsLocationFrozen,
  readOnly,
}) => {
  const readOnlyFromAddressOrSiren = formIsLocationFrozen || fieldReadOnlyBecauseFrozenFormSiret
  const fieldReadOnly = readOnly || readOnlyFromAddressOrSiren

  return (
    <div className="section">
      <h2 className="main-list-title">ADRESSE</h2>
      <div className="field-group">
        <HiddenField name="isLocationFrozen" />
        <AddressField
          form={form}
          label="NumÃ©ro et voie : "
          latitude={formLatitude}
          longitude={formLongitude}
          name="address"
          readOnly={readOnly || fieldReadOnlyBecauseFrozenFormSiret}
          required
          withMap
        />
        <TextField
          autoComplete="postal-code"
          innerClassName="col-33"
          label="Code postal : "
          name="postalCode"
          readOnly={fieldReadOnly}
          required
        />
        <TextField
          autoComplete="address-level2"
          innerClassName="col-66"
          label="Ville : "
          name="city"
          readOnly={fieldReadOnly}
          required
        />
        <NumberField
          innerClassName="col-33"
          label="Latitude : "
          name="latitude"
          readOnly={fieldReadOnly}
          required
        />
        <NumberField
          innerClassName="col-33"
          label="Longitude : "
          name="longitude"
          readOnly={fieldReadOnly}
          required
        />
      </div>
    </div>
  )
}

LocationFields.defaultProps = {
  fieldReadOnlyBecauseFrozenFormSiret: false,
  formIsLocationFrozen: false,
  formLatitude: null,
  formLongitude: null,
  readOnly: true,
}

LocationFields.propTypes = {
  fieldReadOnlyBecauseFrozenFormSiret: PropTypes.bool,
  form: PropTypes.object.isRequired,
  formIsLocationFrozen: PropTypes.bool,
  formLatitude: PropTypes.number,
  formLongitude: PropTypes.number,
  readOnly: PropTypes.bool,
}

export default LocationFields
