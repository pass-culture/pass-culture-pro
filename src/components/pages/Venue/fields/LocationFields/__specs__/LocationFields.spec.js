import { shallow } from 'enzyme'
import React from 'react'
import LocationFields from '../LocationFields'
import { HiddenField } from '../../../../../layout/form/fields'

describe('src | components | pages | Venue | fields | LocationFields', () => {
  let props

  beforeEach(() => {
    props = {
      fieldReadOnlyBecauseFrozenFormSiret: true,
      form: {},
      formIsLocationFrozen: true,
      formLatitude: 1,
      formLongitude: 1,
      readOnly: true
    }
  })

  it('should match snapshot', () => {
    // when
    const wrapper = shallow(<LocationFields {...props}/>)

    // then
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  describe('render', () => {
    it('should render an HiddenField component with the right props', () => {
      // when
      const wrapper = shallow(<LocationFields {...props}/>)

      // then
      const hiddenField = wrapper.find(HiddenField)
      expect(hiddenField).toHaveLength(1)
      expect(hiddenField.prop('name')).toBe('isLocationFrozen')
    })
  })
})
