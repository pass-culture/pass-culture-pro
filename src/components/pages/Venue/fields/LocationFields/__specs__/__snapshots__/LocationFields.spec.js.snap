// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | Venue | fields | LocationFields should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LocationFields
    fieldReadOnlyBecauseFrozenFormSiret={true}
    form={Object {}}
    formIsLocationFrozen={true}
    formLatitude={1}
    formLongitude={1}
    readOnly={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="main-list-title"
        >
          ADRESSE
        </h2>,
        <div
          className="field-group"
        >
          <Unknown
            name="isLocationFrozen"
            validator={[Function]}
          />
          <AddressField
            className=""
            disabled={false}
            form={Object {}}
            format={null}
            id={null}
            innerClassName={null}
            label="Numéro et voie : "
            latitude={1}
            longitude={1}
            name="address"
            parse={null}
            placeholder=""
            readOnly={false}
            required={true}
            type="text"
            validate={null}
            withMap={true}
          />
          <TextField
            autoComplete="postal-code"
            className=""
            disabled={false}
            format={null}
            id={null}
            innerClassName="col-33"
            label="Code postal : "
            name="postalCode"
            parse={null}
            placeholder=""
            readOnly={true}
            renderInner={[Function]}
            renderValue={[Function]}
            required={true}
            type="text"
            validate={null}
          />
          <TextField
            autoComplete="address-level2"
            className=""
            disabled={false}
            format={null}
            id={null}
            innerClassName="col-66"
            label="Ville : "
            name="city"
            parse={null}
            placeholder=""
            readOnly={true}
            renderInner={[Function]}
            renderValue={[Function]}
            required={true}
            type="text"
            validate={null}
          />
          <Unknown
            innerClassName="col-33"
            label="Latitude : "
            name="latitude"
            readOnly={true}
            required={true}
          />
          <Unknown
            innerClassName="col-33"
            label="Longitude : "
            name="longitude"
            readOnly={true}
            required={true}
          />
        </div>,
      ],
      "className": "section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ADRESSE",
          "className": "main-list-title",
        },
        "ref": null,
        "rendered": "ADRESSE",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              name="isLocationFrozen"
              validator={[Function]}
            />,
            <AddressField
              className=""
              disabled={false}
              form={Object {}}
              format={null}
              id={null}
              innerClassName={null}
              label="Numéro et voie : "
              latitude={1}
              longitude={1}
              name="address"
              parse={null}
              placeholder=""
              readOnly={false}
              required={true}
              type="text"
              validate={null}
              withMap={true}
            />,
            <TextField
              autoComplete="postal-code"
              className=""
              disabled={false}
              format={null}
              id={null}
              innerClassName="col-33"
              label="Code postal : "
              name="postalCode"
              parse={null}
              placeholder=""
              readOnly={true}
              renderInner={[Function]}
              renderValue={[Function]}
              required={true}
              type="text"
              validate={null}
            />,
            <TextField
              autoComplete="address-level2"
              className=""
              disabled={false}
              format={null}
              id={null}
              innerClassName="col-66"
              label="Ville : "
              name="city"
              parse={null}
              placeholder=""
              readOnly={true}
              renderInner={[Function]}
              renderValue={[Function]}
              required={true}
              type="text"
              validate={null}
            />,
            <Unknown
              innerClassName="col-33"
              label="Latitude : "
              name="latitude"
              readOnly={true}
              required={true}
            />,
            <Unknown
              innerClassName="col-33"
              label="Longitude : "
              name="longitude"
              readOnly={true}
              required={true}
            />,
          ],
          "className": "field-group",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "name": "isLocationFrozen",
              "validator": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "disabled": false,
              "form": Object {},
              "format": null,
              "id": null,
              "innerClassName": null,
              "label": "Numéro et voie : ",
              "latitude": 1,
              "longitude": 1,
              "name": "address",
              "parse": null,
              "placeholder": "",
              "readOnly": false,
              "required": true,
              "type": "text",
              "validate": null,
              "withMap": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "autoComplete": "postal-code",
              "className": "",
              "disabled": false,
              "format": null,
              "id": null,
              "innerClassName": "col-33",
              "label": "Code postal : ",
              "name": "postalCode",
              "parse": null,
              "placeholder": "",
              "readOnly": true,
              "renderInner": [Function],
              "renderValue": [Function],
              "required": true,
              "type": "text",
              "validate": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "autoComplete": "address-level2",
              "className": "",
              "disabled": false,
              "format": null,
              "id": null,
              "innerClassName": "col-66",
              "label": "Ville : ",
              "name": "city",
              "parse": null,
              "placeholder": "",
              "readOnly": true,
              "renderInner": [Function],
              "renderValue": [Function],
              "required": true,
              "type": "text",
              "validate": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "innerClassName": "col-33",
              "label": "Latitude : ",
              "name": "latitude",
              "readOnly": true,
              "required": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "innerClassName": "col-33",
              "label": "Longitude : ",
              "name": "longitude",
              "readOnly": true,
              "required": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="main-list-title"
          >
            ADRESSE
          </h2>,
          <div
            className="field-group"
          >
            <Unknown
              name="isLocationFrozen"
              validator={[Function]}
            />
            <AddressField
              className=""
              disabled={false}
              form={Object {}}
              format={null}
              id={null}
              innerClassName={null}
              label="Numéro et voie : "
              latitude={1}
              longitude={1}
              name="address"
              parse={null}
              placeholder=""
              readOnly={false}
              required={true}
              type="text"
              validate={null}
              withMap={true}
            />
            <TextField
              autoComplete="postal-code"
              className=""
              disabled={false}
              format={null}
              id={null}
              innerClassName="col-33"
              label="Code postal : "
              name="postalCode"
              parse={null}
              placeholder=""
              readOnly={true}
              renderInner={[Function]}
              renderValue={[Function]}
              required={true}
              type="text"
              validate={null}
            />
            <TextField
              autoComplete="address-level2"
              className=""
              disabled={false}
              format={null}
              id={null}
              innerClassName="col-66"
              label="Ville : "
              name="city"
              parse={null}
              placeholder=""
              readOnly={true}
              renderInner={[Function]}
              renderValue={[Function]}
              required={true}
              type="text"
              validate={null}
            />
            <Unknown
              innerClassName="col-33"
              label="Latitude : "
              name="latitude"
              readOnly={true}
              required={true}
            />
            <Unknown
              innerClassName="col-33"
              label="Longitude : "
              name="longitude"
              readOnly={true}
              required={true}
            />
          </div>,
        ],
        "className": "section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ADRESSE",
            "className": "main-list-title",
          },
          "ref": null,
          "rendered": "ADRESSE",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Unknown
                name="isLocationFrozen"
                validator={[Function]}
              />,
              <AddressField
                className=""
                disabled={false}
                form={Object {}}
                format={null}
                id={null}
                innerClassName={null}
                label="Numéro et voie : "
                latitude={1}
                longitude={1}
                name="address"
                parse={null}
                placeholder=""
                readOnly={false}
                required={true}
                type="text"
                validate={null}
                withMap={true}
              />,
              <TextField
                autoComplete="postal-code"
                className=""
                disabled={false}
                format={null}
                id={null}
                innerClassName="col-33"
                label="Code postal : "
                name="postalCode"
                parse={null}
                placeholder=""
                readOnly={true}
                renderInner={[Function]}
                renderValue={[Function]}
                required={true}
                type="text"
                validate={null}
              />,
              <TextField
                autoComplete="address-level2"
                className=""
                disabled={false}
                format={null}
                id={null}
                innerClassName="col-66"
                label="Ville : "
                name="city"
                parse={null}
                placeholder=""
                readOnly={true}
                renderInner={[Function]}
                renderValue={[Function]}
                required={true}
                type="text"
                validate={null}
              />,
              <Unknown
                innerClassName="col-33"
                label="Latitude : "
                name="latitude"
                readOnly={true}
                required={true}
              />,
              <Unknown
                innerClassName="col-33"
                label="Longitude : "
                name="longitude"
                readOnly={true}
                required={true}
              />,
            ],
            "className": "field-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "name": "isLocationFrozen",
                "validator": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "disabled": false,
                "form": Object {},
                "format": null,
                "id": null,
                "innerClassName": null,
                "label": "Numéro et voie : ",
                "latitude": 1,
                "longitude": 1,
                "name": "address",
                "parse": null,
                "placeholder": "",
                "readOnly": false,
                "required": true,
                "type": "text",
                "validate": null,
                "withMap": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoComplete": "postal-code",
                "className": "",
                "disabled": false,
                "format": null,
                "id": null,
                "innerClassName": "col-33",
                "label": "Code postal : ",
                "name": "postalCode",
                "parse": null,
                "placeholder": "",
                "readOnly": true,
                "renderInner": [Function],
                "renderValue": [Function],
                "required": true,
                "type": "text",
                "validate": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoComplete": "address-level2",
                "className": "",
                "disabled": false,
                "format": null,
                "id": null,
                "innerClassName": "col-66",
                "label": "Ville : ",
                "name": "city",
                "parse": null,
                "placeholder": "",
                "readOnly": true,
                "renderInner": [Function],
                "renderValue": [Function],
                "required": true,
                "type": "text",
                "validate": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "innerClassName": "col-33",
                "label": "Latitude : ",
                "name": "latitude",
                "readOnly": true,
                "required": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "innerClassName": "col-33",
                "label": "Longitude : ",
                "name": "longitude",
                "readOnly": true,
                "required": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
